hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/StateContext.java

```diff
@@ -532,12 +532,14 @@ public List<ContainerAction> getPendingContainerAction(
    *
    * @param pipelineAction PipelineAction to be added
    */
-  public void addPipelineActionIfAbsent(PipelineAction pipelineAction) {
+  public boolean addPipelineActionIfAbsent(PipelineAction pipelineAction) {
     // Put only if the pipeline id with the same action is absent.
     final PipelineKey key = new PipelineKey(pipelineAction);
+    boolean added = false;
     for (InetSocketAddress endpoint : endpoints) {
-      pipelineActions.get(endpoint).putIfAbsent(key, pipelineAction);
+      added = added || pipelineActions.get(endpoint).putIfAbsent(key, pipelineAction);
```

Comment:
To keep existing behavior, `||` condition should not be short-circuited.

```suggestion
      added = pipelineActions.get(endpoint).putIfAbsent(key, pipelineAction) || added;
```